libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,120 @@
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <cmath>
+
+#include <catch2/catch.hpp>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/analysis/update.hpp>
+
+namespace analysis {
+void run_analysis_update_without_rowscaling(
+    analysis_module_type *module, const bool_vector_type *ens_mask,
+    const meas_data_type *forecast, obs_data_type *obs_data,
+    rng_type *shared_rng, matrix_type *E,
+    std::unordered_map<std::string, matrix_type *> parameters);
+}
+
+const double a_true = 5.0;
+const double b_true = 0.5;
+
+struct model {
+    double a;
+    double b;
+
+    model(double a, double b) : a(a), b(b) {}
+
+    model(rng_type *rng) {
+        this->a = enkf_util_rand_normal(a_true, 0.5, rng);
+        this->b = enkf_util_rand_normal(b_true, 0.5, rng);
+    }
+
+    double eval(double x) const { return this->a * x + this->b; }
+};
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+    GIVEN("Increasing ensemble size") {
+        int ens_size = GENERATE(10, 100, 1000);
+        int obs_size = 2;
+
+        std::vector<model> ens;
+
+        analysis_module_type *ies_module = analysis_module_alloc("IES_ENKF");
```

Comment:
We are going in a direction where you can use the `IES_ENKF` module as done in this test, but not yet. The `IES_ENKF` module is by default an iterable module which should be used in an iterable fashion, it will be incorrectly used in the current testing scenario. For this particular test I would strongly recommend to use the `STD_ENKF` module.